common/initial_channel.h

```diff
@@ -114,6 +116,27 @@ static inline u32 htlc_minimum_msat(const struct channel *channel,
 {
 	return channel->config[recipient]->htlc_minimum_msat;
 }
+/*
+ * The maximum msat that this node will accept for an htlc.
+ * It's flagged as an optional field in `channel_update`.
+ *
+ * We advertise the maximum value possible,
+ * defined as the smaller of either the remote's indicated
+ * maximum in-flight HTLC, our max allowed in-flight HTLC,
+ * or the total channel capacity minus the cumulative reserve.
+ * FIXME: does this need fuzz?
+ * FIXME: where to put this?
+ */
+static inline u64 htlc_maximum_msat(const u64 funding_msat,
+		const struct channel_config *our_config,
+		const struct channel_config *remote_config)
+{
+	u64 min_htlc_msat = min(our_config->max_htlc_value_in_flight_msat,
+			remote_config->max_htlc_value_in_flight_msat);
```

Comment:
Interestingly, our_config->max_htlc_value_in_flight_msat doesn't matter; that effects only incoming HTLCs, not outgoing ones (which is what this channel_update is about).

Reply:
I think this belongs in channeld.c, perhaps called 'advertized_htlc_max()'?